Lib/test/test_typing.py

```diff
@@ -2535,6 +2535,100 @@ def meth(x): ...
         with self.assertRaises(TypeError):
             isinstance(C(), BadPG)
 
+    def test_protocols_isinstance_properties_and_descriptors(self):
+        class C:
+            @property
+            def attr(self):
+                return 42
```

Comment:
What about side-effects in `@property` like `raise ValueError`? Should we test this case?

Reply:
We should test this case, definitely! But the behaviour for this case may be about to change if one of the patches discussed in https://github.com/python/typing/issues/1363 is implemented (and the consensus seems to be that we should implement one of those patches). Whether or not we decide to change the behaviour around properties that have side effects, I'd prefer to add those tests in a separate PR.